python/cugraph-dgl/cugraph_dgl/_version.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import importlib.resources
+
+__version__ = (
+    importlib.resources.files("cugraph_dgl").joinpath("VERSION").read_text().strip()
```

Comment:
I think this was mentioned in another PR for this in another repo, but I'm also not crazy about reading another file on import, but I think I understand the reasoning.  Can there be a comment above these lines explaining why this is happening, and summarize what's it's doing?
Also, it looks like `VERSION` redirects to `../../../VERSION`.  Does this mean when the package is installed (without something like `pip -e`) there's a file named `VERSION` in `site-packages` (or somewhere similar)?

Reply:
I'm curious if there's an advantage to having this file read `VERSION` in its package which then redirects to the repo `../../../VERSION` as opposed to just having this file read `../../../VERSION`? Or maybe I'm missing something.

Reply:
The symlink is necessary because importlib reads resources at the module level. For the multitude of ways that we can install a python package, we can only guarantee that the module level VERSION file will be available.

Reply:
It's also worth noting that when the package is actually built, setuptools will copy the original files for symlinks pointing outside the package, so if you download a conda package or a wheel this `VERSION` file won't be a symlink anymore, it'll be a copy of the file.

Reply:
> It's also worth noting that when the package is actually built, setuptools will copy the original files for symlinks pointing outside the package, so if you download a conda package or a wheel this `VERSION` file won't be a symlink anymore, it'll be a copy of the file.

Ah, I suspected something like that but I think this comment was the missing piece. Thanks!